### PR TITLE
Adding pinch zoom config option.

### LINKED ISSUE #2782: Pinch to zoom can't be configured / turned off
**Describe the bug**

Pinch to zoom doesn't respect zoom options in order to figure out whether or not to zoom the workspace. 

**To Reproduce**

1. Configure the playground without any zoom options.
2. Browse to the playground on a touch device. 
3. Try to pinch to zoom the workspace
4. It works, when it shouldn't..

**Expected behavior**

I expect a developer can turn off pinch to zoom with the zoom options.

**Smartphone (please complete the following information):**
 - Device: Iphone X
 - OS: IOS 12
 - Browser: Safari

**Additional context**

Currently there is no zoom option for pinch to zoom. We have a few options: 
- We could overload wheel. 
- We could create a new option.
- We could just check for the existence of the zoom option (ie: any zoom option, enables it). 

I feel if we create a zoom option for it now, we'd have to communicate to people that they have to now opt into it through that option.

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

#2782

### Proposed Changes

Adds pinch zoom option.

### Reason for Changes

Allows for pinch zoom to be configured.

### Test Coverage

Tested on:
* Desktop Chrome

### CODE REVIEW

File: core/workspace_svg.js [hunk 1]
```diff
@@ -1811,12 +1813,6 @@ Blockly.WorkspaceSvg.prototype.setBrowserFocus = function() {
  *     amount values zoom in.
  */
 Blockly.WorkspaceSvg.prototype.zoom = function(x, y, amount) {
-  // TODO (#2782): Consider removing once pinch understands zoom configuration
```
Comment (moniika) [inline core/workspace_svg.js]: Currently testing mutators.

File: core/options.js [hunk 2]
```diff
@@ -247,6 +247,11 @@ Blockly.Options.parseZoomOptions_ = function(options) {
   } else {
     zoomOptions.scaleSpeed = Number(zoom['scaleSpeed']);
   }
+  if (zoom['pinch'] === undefined) {
+    zoomOptions.pinch = zoomOptions.wheel || zoomOptions.controls;
+  } else {
+    zoomOptions.pinch =!! zoom['pinch'];
```
[the remark this answers is not in the document]
Comment (moniika) [reply core/options.js]: Done.

File: core/workspace_svg.js (hunk 1 above)
Comment (moniika) [reply core/workspace_svg.js]: tested with breakpoints.

**Comment by RoboErikG** [comment]
@moniika is this ready for another review pass?

**Comment by samelhusseini** [comment]
This mostly looks good, I would just cache the whether or not to handlePinch in the constructor of the TouchGesture object.

File: core/touch_gesture.js [hunk 3]
```diff
@@ -256,33 +256,47 @@ Blockly.TouchGesture.prototype.handleTouchMove = function(e) {
   // Update the cache
   this.cachedPoints_[pointerId] = this.getTouchPoint(e);
 
+  var workspace = this.startWorkspace_;
   var pointers = Object.keys(this.cachedPoints_);
-  // If two pointers are down, check for pinch gestures
-  if (pointers.length == 2) {
-    // Calculate the distance between the two pointers
-    var point0 = /** @type {!Blockly.utils.Coordinate} */ (
+  if (workspace.options.zoomOptions && workspace.options.zoomOptions.pinch &&
```
[the remark this answers is not in the document]
Comment (moniika) [reply core/touch_gesture.js]: My concern with caching is that it wouldn't be reflected when the value changes, but I'll take a look at where I could potentially add the cache and see what it looks like.

[the remark this answers is not in the document]
Comment (samelhusseini) [reply core/touch_gesture.js]: I don't believe we supporting changing workspace options on the fly.

[the remark this answers is not in the document]
Comment (moniika) [reply core/touch_gesture.js]: Good to know, it also looks like new gesture is created every time.

File: core/touch_gesture.js [hunk 4]
```diff
@@ -257,32 +265,44 @@ Blockly.TouchGesture.prototype.handleTouchMove = function(e) {
...
+    this.handlePinch(e);
+  } else {
+    Blockly.TouchGesture.superClass_.handleMove.call(this, e);
+  }
+};
+
+/**
+* Handle pinch zoom gesture.
+* @param {!Event} e A touch move, or pointer move event.
+* @package
```
Comment (samelhusseini) [inline core/touch_gesture.js]: Why is this @package? And not private

File: core/touch_gesture.js [hunk 5]
```diff
@@ -257,32 +265,44 @@ Blockly.TouchGesture.prototype.handleTouchMove = function(e) {
...
+    this.cachedPoints_[pointers[1]]);
+  var moveDistance = Blockly.utils.Coordinate.distance(point0, point1);
+  var startDistance = this.startDistance_;
+  this.touchScale_ = moveDistance / startDistance;
 
-    if (this.previousScale_ > 0 && this.previousScale_ < Infinity) {
-      var gestureScale = scale - this.previousScale_;
-      var delta = gestureScale > 0 ?
+  if (this.previousScale_ > 0 && this.previousScale_ < Infinity) {
+    var gestureScale = this.touchScale_ - this.previousScale_;
```
Comment (samelhusseini) [inline core/touch_gesture.js]: Curious why scale is now stored on the object? I don’t see it read anywhere outside this method

File: core/touch_gesture.js (hunk 4 above)
Comment (moniika) [reply core/touch_gesture.js]: Could be private, changing it now.

File: core/touch_gesture.js (hunk 5 above)
Comment (moniika) [reply core/touch_gesture.js]: It was always stored on the object (see old L269). Given its a private field and I also don't see it anywhere outside of this method, I'm going to change it to a local variable and if we need it in future, we can update it then.

**Pull request merged**